main.go

```diff
@@ -211,6 +256,15 @@ func registerMetrics() {
 	prometheus.MustRegister(promLatencyHistogram)
 }
 
+// Sample Rate is between [0.0, 1.0] and determines what percentage of request bodies
+// should be checked that their hash matches a known hash.
+// This works best if the input ids are uniform. (so like incremented request ids)
+func shouldCheckHash(sampleRate float64, id uint64) bool {
+	v := sampleRate * 10000.0
+	t := math.Mod(float64(id), 10000.0)
+	return v > t
+}
```

Comment:
The drawback of calculating this as a mod of the request id means that the checks aren't evenly distributed across each block of 10000 requests. For instance, if I'm sending 1000 requests per second with a 0.1 sample rate, then all of the requests in the first second will be checked, and then there will be 9 seconds with no checks before we check again. I'd rather see this done randomly per request based on the sample rate, so that some checks happen in each second. Something like:

```
func shouldCheckHash(sampleRate float64) bool {
	return rand.Float64() < sampleRate
}
```

It's slightly less precise, but I think evenly distributing is key.